core/src/Kokkos_HPX.hpp

```diff
@@ -523,17 +525,74 @@ namespace Kokkos {
 namespace Experimental {
 template <>
 class UniqueToken<HPX, UniqueTokenScope::Instance> {
+ private:
+  using buffer_type = Kokkos::View<uint32_t *, Kokkos::HostSpace>;
+  int m_count;
+  buffer_type m_buffer_view;
+  uint32_t volatile *m_buffer;
+
  public:
   using execution_space = HPX;
   using size_type       = int;
-  UniqueToken(execution_space const & = execution_space()) noexcept {}
 
-  // NOTE: Currently this assumes that there is no oversubscription.
-  // hpx::get_num_worker_threads can't be used directly because it may yield
-  // it's task (problematic if called after hpx::get_worker_thread_num).
-  int size() const noexcept { return HPX::impl_max_hardware_threads(); }
-  int acquire() const noexcept { return HPX::impl_hardware_thread_id(); }
-  void release(int) const noexcept {}
+  /// \brief create object size for concurrency on the given instance
+  ///
+  /// This object should not be shared between instances
```

Comment:
I have trouble parsing this comment.  Please improve here and at places it was copied from.